src/test/endpoints.test.js

```diff
@@ -0,0 +1,69 @@
+const main = require('../../main');
+const original_fetch = require('node-fetch');
+const models = require('../../app/models');
+
+const port = process.env.PORT || 8888;
+const base = 'http://localhost:' + port;
+const api = path => base + (path.startsWith('/') ? '' : '/') + path;
+
+class FetchError extends Error {
+    constructor(response, msg = 'We did not get a 2xx response code') {
+        super(msg);
+        this.response = response;
+    }
+}
+
+const fetch = (uri, options = {}) => {
+    const headers = { 'Content-Type': 'application/json', ... options.headers };
+    options = { ... options, headers };
+
+    if ( options.hasOwnProperty('body') ) {
+        options.body = JSON.stringify(options.body);
+    }
+
+    let result = original_fetch(api(uri), options);
+    if ( options.raw ) { return result; }
+
+    return result.then(res => {
+        if ( res.ok ) { return res.json(); }
+        else { throw new FetchError(res); }
+    });
+};
+
+const post = (uri, body, options = {}) => {
+    return fetch(uri, { ... options, method: 'post', body });
+};
+
+describe('users endpoint', () => {
+
+    let server = null;
+    beforeAll(() => main(port).then(s => { server = s; }));
+    afterAll(() => server && server.close());
```

Comment:
If i'm correct this will start a single instance of the server for all tests. Not one for each individual test. This means only 1 database instance will be created (i believe the db instance is created in memory at server start time). Doesn't that cause test leakage?

Reply:
Yes there is only one server instance, and only one database instance. But if you look closely at the code, you'll notice that the sequelize initialization only runs once, no matter how many times you start the server. 

But there is another method that resets the database before each test

```javascript
beforeEach(() => models.User.truncate());
```

I think this will do the something similar to running `truncate table \`users`\;` on that table. This will prevent leakage (as long as tests are not run in parallel.